src/components/WrapEtherBtn.tsx

```diff
@@ -222,14 +244,38 @@ const WrapUnwrapEtherBtn: React.FC<WrapUnwrapEtherBtnProps> = (props: WrapUnwrap
                   required
                   ref={register({
                     pattern: { value: validInputPattern, message: 'Invalid amount' },
-                    validate: { positive: validatePositiveConstructor('Invalid amount') },
+                    validate: {
+                      positive: validatePositiveConstructor('Invalid amount'),
+                      max: (value: string): string | true => {
+                        const amount = parseAmount(value, DEFAULT_PRECISION) || ZERO
+
+                        if (balance && amount.gt(balance)) {
+                          // Not enough balance
+                          return "The amount cannot be bigger than what's available"
+                        } else {
+                          // Enough balance
+                          return true
+                        }
+                      },
+                    },
                     required: 'The amount is required',
                     min: 0,
                   })}
                 />
               </InputBox>
             </div>
-            {amountError && <p className="error">Invalid amount</p>}
+            {amountError && <p className="error">{amountError.message}</p>}
```

Comment:
really not comfortable with wrap all, it's a recipe for disaster and has overwhelmingly worse implications than benefits if done incorrectly even once.

I would scrap it and see it as an extra security layer that users themselves have to consciously put in the amount max themselves to verify. it's like when u delete a repo and have to type in the name of it before being able to submit

Reply:
I'm happy to remove the link to autofill the amount

I guess the warning it still make sense right? @W3stside , @Velenir , @alfetopito

Reply:
8a477fe7bf86eaa043913875bb7260337fa1ec48

Reply:
Closing, let reopen if u still have comments on the new approach